### DISCUSSION TITLE
Expected failure of daily Cypress job

### DISCUSSION THREAD

**Discussion by MikeMcC399** [corona-warn-app/cwa-website#3557] [Development] [answered]
GitHub is currently rolling out the runner image [ubuntu-22.04 (20230917)](https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230917.1) which contains Google Chrome `117.0.5938.88`.

Google Chrome `117.0.5938.88` contains Bug [1483163](https://bugs.chromium.org/p/chromium/issues/detail?id=1483163) which means it will crash in headless mode when run with Cypress version `12.14.0` and earlier.

The `cwa-website` repository is configured with Cypress `12.8.1` and so it is impacted. This includes the GitHub Actions workflow [cypress-test-prod.yml](https://github.com/corona-warn-app/cwa-website/blob/master/.github/workflows/cypress-test-prod.yml) which runs daily.

During rollout GitHub randomly provides either the new runner image or the old runner image, so it is unpredictable during this time whether the job will fail. So far it has not failed because it has used the old image [ubuntu-22.04 (20230911)](https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230911.1) - see for instance run [6259601072](https://github.com/corona-warn-app/cwa-website/actions/runs/6259601072)
```text
Runner Image
  Image: ubuntu-22.04
  Version: 20230911.1.0
```

## Options

1. Do nothing. The daily workflow will soon start failing and other workflows will fail if triggered through new PRs. 

    Chromium has fixed the bug in the Chrome `119` Canary release and plans to back-port it to Chrome `117`. It may take 2 to 4 weeks before the fix is published and rolled out to GitHub Actions. When the fixed has been rolled out the workflows can be expected to run without failing again.

2. Update Cypress to version `12.15.0` or later. 

    It seems that the Cypress tests in this repository are compatible with the current latest version of Cypress `13.2.0`.

## Note

I'm just passing on this information since I am working a lot with Cypress these days. I'm not intending to submit any PR here since I already stopped contributing changes to CWA and I no longer maintain a fork of the repo.

**Comment by dsarkar** [comment, marked as answer]
@MikeMcC399 Thanks for passing this information! Very much appreciated; we will have a look at this!

**Reply by MikeMcC399** [reply]
@dsarkar 

It's good to hear a voice from the old team!

As I expected, today the workflow failed in job https://github.com/corona-warn-app/cwa-website/actions/runs/6272357319:

```text
  Running:  app_to_web.cy.js                                                                (1 of 8)
Still waiting to connect to Chrome, retrying in 1 second (attempt 18/62)
read ECONNRESET
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20)
Error: Process completed with exit code 1.
```

Since the workflow https://github.com/corona-warn-app/cwa-website/actions/workflows/cypress-test-prod.yml was enabled by @svengabr, he will be the one to receive an error e-mail. If that does not fit the current organisation then you could disable and re-enable the workflow to get it assigned to somebody else.

**Reply by dsarkar** [reply]
Good morning @MikeMcC399,

Thanks again for your suggestions. For the moment I have disabled the workflow:
https://github.com/corona-warn-app/cwa-website/actions/workflows/cypress-test-prod.yml

Next week we will have a closer look and check if we update Cypress.

Enjoy your weekend! Best, DS

**Reply by MikeMcC399** [reply]
@dsarkar 

You're welcome!

Have a nice weekend yourself!

**Reply by MikeMcC399** [reply]
@dsarkar 

The bug is fixed in Chrome Version `117.0.5938.132` (Official Build) (64-bit) for Linux. I guess that this will roll on on GitHub in a week or two.

**Reply by MikeMcC399** [reply]
Google Chrome [Stable Channel Update for Desktop](https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_27.html)
> Wednesday, September 27, 2023
The Stable channel has been updated to `117.0.5938.132` for Windows, Mac and Linux, which will roll out over the coming days/weeks.

**Reply by MikeMcC399** [reply]
@dsarkar 

The new runner image [Ubuntu2204 - 20231001.1.0](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md) includes the patch Google Chrome `117.0.5938.132`. It is currently 99.98% rolled out.

**Reply by MikeMcC399** [reply]
You can re-enable the workflow now and expect it to succeed.

**Reply by larswmh** [reply]
Good morning @MikeMcC399,

thanks for the information. I've re-enabled the workflow now.

**Reply by MikeMcC399** [reply]
Hello @larswmh 

Good to hear from you again 🙂  and to see that the workflow was successful [run 6454646888](https://github.com/corona-warn-app/cwa-website/actions/runs/6454646888) 🎉 😄 !

**Comment by MikeMcC399** [comment]
Issue is fixed.
